base/server/src/main/java/com/netscape/cmscore/dbs/CertRecord.java

```diff
@@ -195,7 +176,7 @@ public void set(String name, Object obj) throws EBaseException {
         } else if (name.equalsIgnoreCase(ATTR_REVOKED_ON)) {
             mRevokedOn = (Date) obj;
         } else {
-            throw new EBaseException(CMS.getUserMessage("CMS_BASE_INVALID_ATTRIBUTE", name));
+            throw new EBaseException(CMS.getUserMessage(CMS_BASE_INVALID_ATTRIBUTE, name));
```

Comment:
I'm guessing that Sonar complained about the duplicate literals. There are around ~1500 calls to `CMS.getUserMessage()` like this, so if we fix it by creating local constants we might end up with a lot of duplicate constants scattered in many places. I think it would be better to consolidate the constants for log messages into a single class, but I'm not sure exactly how that would work since this might be related to localization.

Reply:
Yeah I had a similar thought, but wasn't sure what to do about it. I'll leave as is for now.